source/module_elecstate/test/CMakeLists.txt

```diff
@@ -2,11 +2,68 @@ remove_definitions(-D__MPI)
 remove_definitions(-D__EXX)
 remove_definitions(-D__CUDA)
 remove_definitions(-D__ROCM)
+add_definitions(-D__NORMAL)
 
 AddTest(
   TARGET EState_updaterhok_pw
-  LIBS ${math_libs} base 
-  SOURCES updaterhok_pw_test.cpp ../../src_pw/charge.cpp ../../src_parallel/parallel_reduce.cpp ../../src_parallel/ft.cpp ../../src_pw/pw_basis_old.cpp ../../src_pw/pw_complement.cpp ../../src_pw/klist.cpp ../../src_parallel/parallel_kpoints.cpp ../../src_pw/occupy.cpp  ../../module_elecstate/elecstate_pw.cpp ../../module_elecstate/elecstate.cpp
+  LIBS ${math_libs} planewave_serial base_serial
+  SOURCES updaterhok_pw_test.cpp ../../src_pw/charge.cpp ../../src_parallel/parallel_reduce.cpp 
+          ../../src_parallel/ft.cpp ../../src_pw/pw_basis_old.cpp ../../src_pw/pw_complement.cpp 
+          ../../src_pw/klist.cpp ../../src_parallel/parallel_kpoints.cpp ../../src_pw/occupy.cpp  
+          ../../module_elecstate/elecstate_pw.cpp ../../module_elecstate/elecstate.cpp
+)
+add_library(
+    base_serial
+    OBJECT
```

Comment:
What's the point of creating another lib for base? For removing MPI-related part?

Reply:
Because unittest in module_elecstate, src_pw, src_lcao uses base library. However, unit test remove MPI and base library should remove MPI. I do not want copy same *.cpp to each unittest. Thus base_serial is added only if TESTING is on.